docs/proposals/20220209-machinepool-machines.md

```diff
@@ -0,0 +1,196 @@
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story U1](#story-u1)
+      - [Story U2](#story-u2)
+      - [Story U3](#story-u3)
+    - [Requirements](#requirements)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+* **Machine**: a declarative specification for an infrastructure component that hosts a Kubernetes Node. Commonly, a Machine represents a VM instance.
+* **MachinePool**: a spec for an infrastructure component that can scale a group of Machines. A MachinePool acts similarly to a [MachineDeployment][], but leverages scaling up resources that are specific to an infrastructure provider. However, while individual Machines in a MachineDeployment are visible and can be removed, individual Machines in a MachinePool are not represented.
+* **MachinePool Machine**: a Machine that is owned by a MachinePool. It requires no special treatment in Cluster API in general; a Machine is a Machine without regard to ownership.
+
+See also the [Cluster API Book Glossary][].
+
+## Summary
+
+MachinePools should be enhanced to own Machines that represent each of its `Replicas`.
+
+These "MachinePool Machines" will open up the following opportunities:
+
+- Share common behavior for Machines in MachinePools, which otherwise would need to be implemented by each infrastructure provider.
+- Allow both Cluster API-driven and externally driven interactions with individual Machines in a MachinePool.
+- Enable MachinePools to use deployment strategies similarly to MachineDeployments.
+
+## Motivation
+
+MachineDeployments and Machines provide solutions for common scenarios such as deployment strategy, machine health checks, and retired node deletion. Exposing MachinePool Machines at the CAPI level will enable similar integrations with external projects. 
+
+For example, Cluster Autoscaler needs to delete individual Machines. Without a representation in the CAPI API of individual Machines in MachinePools, projects building integrations with Cluster API would need to be specialized to delete the cloud-specific resources for each MachinePool infrastructure provider. By exposing MachinePool Machines in CAPI, Cluster Autoscaler can treat Machines built by MachineDeployments and MachinePools in a uniform fashion through CAPI APIs.
+
+As another example, currently each MachinePool infrastructure provider must implement their own node drain functionality. When MachinePool Machines become available in the CAPI API, all providers can use its existing node drain implementation.
+
+### Goals
```

Comment:
That's a good question. It probably depends on whether we would stop sharing a single KubeadmConfig for all MachinePool machines or start having a single KubeadmConfigTemplate and then create a new KubeadmConfig for each machine.

> can we move to "short-lived" token for each machines vs having a single token never being deleted and shared between all the machines?

to clarify, we currently don't use a single long-lived token for MachinePools. We rotate the token every 15 minutes and generate a new one so that there is always a fresh valid one ready to be used.

Reply:
thanks for clarification
what about adding two goals with a soft commitment to explore this two topics (better code organization, improved bootstrap token management)